realm/realm-annotations-processor/src/main/java/io/realm/processor/Constants.kt

```diff
@@ -70,34 +74,42 @@ object Constants {
     }
 
     init {
-        JAVA_TO_REALM_TYPES["byte"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["short"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["int"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["long"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["float"] = RealmFieldType.FLOAT
-        JAVA_TO_REALM_TYPES["double"] = RealmFieldType.DOUBLE
-        JAVA_TO_REALM_TYPES["boolean"] = RealmFieldType.BOOLEAN
-        JAVA_TO_REALM_TYPES["java.lang.Byte"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["java.lang.Short"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["java.lang.Integer"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["java.lang.Long"] = RealmFieldType.INTEGER
-        JAVA_TO_REALM_TYPES["java.lang.Float"] = RealmFieldType.FLOAT
-        JAVA_TO_REALM_TYPES["java.lang.Double"] = RealmFieldType.DOUBLE
-        JAVA_TO_REALM_TYPES["java.lang.Boolean"] = RealmFieldType.BOOLEAN
-        JAVA_TO_REALM_TYPES["java.lang.String"] = RealmFieldType.STRING
-        JAVA_TO_REALM_TYPES["java.util.Date"] = RealmFieldType.DATE
-        JAVA_TO_REALM_TYPES["byte[]"] = RealmFieldType.BINARY
-        // TODO: add support for char and Char
+        JAVA_TO_REALM_TYPES.apply {
+            this["byte"] = RealmFieldType.INTEGER
```

Comment:
You could do `JAVA_TO_REALM_TYPES = mapOf("byte" to RealmFieldType.INTEGER)` and so on to make it more Kotlin-y